src/exchange/adl_handler.cairo

```diff
@@ -145,9 +163,17 @@ mod AdlHandler {
             swap_handler_address,
             referral_storage_address
         );
+        self.data_store.write(IDataStoreSafeDispatcher { contract_address: data_store_address });
```

Comment:
Cant we just access this fields from IBaseOrderHandler?

Reply:
Could you clarify what you're asking?

Reply:
![image](https://github.com/keep-starknet-strange/satoru/assets/37303126/2a7488e6-7700-4c52-a7f2-0879f30fedda)

I think something like that could be possible, since data_store is already stored in BaseOrderHandler.

Here data_store and others are stored both in BaseOrderHandler and AdlHandler

Reply:
Given the changes, can we consider removing all constructor parameters from `adl_handler`?

Reply:
Updated the code as requested.